src/core_plugins/kibana/server/routes/api/suggestions/register_value_suggestions.js

```diff
@@ -4,33 +4,64 @@ export function registerValueSuggestions(server) {
   server.route({
     path: '/api/kibana/suggestions/values/{index}',
     method: ['POST'],
-    handler: function (req, reply) {
+    handler: async function (req, reply) {
       const { index } = req.params;
       const { field, query } = req.payload;
-
       const { callWithRequest } = server.plugins.elasticsearch.getCluster('data');
-      const include = query ? `.*${query}.*` : undefined;
-      const body = getBody({
-        field,
-        include,
-        shard_size: 10,
-        size: 10
-      });
-
-      return callWithRequest(req, 'search', { index, body })
-      .then((res) => {
-        const suggestions = res.aggregations.suggestions.buckets.map(bucket => bucket.key);
-        return reply(suggestions);
-      })
-      .catch(error => reply(handleESError(error)));
+      const body = getBody({ field, query });
+      try {
+        const response = await callWithRequest(req, 'search', { index, body });
+        const suggestions = response.aggregations.suggestions.buckets.map(bucket => bucket.key);
+        reply(suggestions);
+      } catch (error) {
+        reply(handleESError(error));
+      }
     }
   });
 }
 
-function getBody(terms) {
+function getBody({ field, query }) {
   return {
+    size: 0,
+    timeout: '1s',
+    query: {
+      query_string: {
+        default_field: field,
+        query: `${getEscapedQuery(query)}*`,
+        analyze_wildcard: true
+      }
+    },
     aggs: {
-      suggestions: { terms }
+      suggestions: {
+        terms: { field }
+      }
```

Comment:
Hey @jpountz, do you see any gotchas with how we're using `timeout` here? We played around with `terminate_after` as you suggested. It helped a lot, but picking a value that would work for all users felt like a shot in the dark. `timeout` is nice because it allows us to control the user experience more explicitly (we can say "suggestions should take no longer than a second"). But I wanted to check with you, is there any reason why we *shouldn't* rely on `timeout` in this way?